Content.Server/Anomaly/Effects/AnomalyCoreSystem.cs

```diff
@@ -0,0 +1,68 @@
+using Content.Server.Anomaly.Components;
+using Content.Server.Cargo.Components;
+using Content.Shared.Anomaly;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Anomaly.Effects;
+
+/// <summary>
+/// This component reduces the value of the entity during decay
+/// </summary>
+public sealed class AnomalyCoreSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<AnomalyCoreComponent, MapInitEvent>(OnMapInit);
+    }
+
+    private void OnMapInit(Entity<AnomalyCoreComponent> core, ref MapInitEvent args)
+    {
+        core.Comp.DecayMoment = _gameTiming.CurTime + TimeSpan.FromSeconds(core.Comp.TimeToDecay);
+
+        if (TryComp<StaticPriceComponent>(core, out var price))
+        {
+            core.Comp.OldPrice = price.Price;
+        }
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<AnomalyCoreComponent>();
+        while (query.MoveNext(out var uid, out var component))
+        {
+            if (component.IsDecayed)
+                continue;
+
+            //checks every few seconds how much time is left to decompose, and decreases the value of the object
+            component.AccumulatedFrametime += frameTime;
+            if (component.AccumulatedFrametime < component.UpdateInterval)
+                continue;
+
+            var timeLeft = component.DecayMoment - _gameTiming.CurTime;
+            var lerp = (double)(timeLeft.TotalSeconds / component.TimeToDecay);
+            lerp = Math.Clamp(lerp, 0, 1);
+
+            if (TryComp<StaticPriceComponent>(uid, out var price))
+                price.Price = MathHelper.Lerp(component.FuturePrice, component.OldPrice, lerp);
```

Comment:
this should absolutely not work by continually updating static price, just remove staticprice on the cores and subscribe to `GetPriceEvent` for anomalycore and calculate it there

Reply:
Done